crates/miden-objects/src/note/note_tag.rs

```diff
@@ -404,53 +413,96 @@ mod tests {
             tag.validate(NoteType::Encrypted)
                 .expect("local execution should support encrypted notes");
         }
+
+        for account_id in network_accounts {
+            let tag = NoteTag::from_account_id(account_id, NoteExecutionMode::Local)
+                .expect("Tag generation must work with local execution and network account ID");
+            assert!(!tag.is_single_target());
+            assert_eq!(tag.execution_mode(), NoteExecutionMode::Local);
+
+            tag.validate(NoteType::Public)
+                .expect("local execution should support public notes");
+            tag.validate(NoteType::Private)
+                .expect("local execution should support private notes");
+            tag.validate(NoteType::Encrypted)
+                .expect("local execution should support encrypted notes");
+        }
     }
 
     #[test]
     fn test_from_account_id_values() {
-        // Private Account ID with the following bit pattern in the first and second byte:
-        // 0b11001100_01010101
-        //   ^^^^^^^^ ^^^^^^  <- these are the 14 bits used in the tag.
+        /// Private Account ID with the following bit pattern in the first and second byte:
+        /// 0b11001100_01010101
+        ///   ^^^^^^^^ ^^^^^^  <- 14 bits of the local tag.
```

Comment:
I _think_ this comment isn't accurate? Or rather the bit pattern doesn't match the bit mask setter. This isn't this PRs fault, looks like it was outdated previously.

This test (and several others) feel like they would benefit from `rstest` and/or splitting into multiple individual tests.

The main benefit is that the test name should identify specifics. Currently there is no way to know whether this test covers all sub-instances/combinations without reading through all of it.